README.md

```diff
@@ -30,18 +29,76 @@ All scripts should be executed in the virtual environment managed by `poetry`.
 poetry run python -m pytest
 ```
 
-### Running the bandit-related scripts
+### Running Experiments
 
-There are three scripts in the bandit_scripts folder:
+Currently, to run an experiment, you should specify three components: `--name` (`-n`), `--simulation_path` (`-s`), and `--algorithm_path` (`-a`).
+The `--name` field is just the name to give to the experiment. 
+Once we set up a Mongo Observer to track experiments with a MongoDB, this will help you track different experiments more efficiently.
+The `--simulation_path` and `--algorithm_path` fields should point to a simulation configuration file and an algorithm configuration file, respectively.
 
-- `show_simulated_subgraph.py` - runs and (optionally) visualizes the simulation (only environment)
-- `show_bandit.py` - runs and (optionally) visualizes the simulation (both agent and environment)
-- `train_bandit.py` - runs the simulation and trains an agent in a given environment (no visualization, logging to TensorBoard)
+**Note:** We do not use default values _anywhere_ in our code.
+Every value must come from the config file.
 
-Each script can be parametrized by arguments passed from the command line. For example, to run the simulation with `ppo` agent in the `noisy_static` environment please run:
+#### Simulation Config
 
-```console
-poetry run python bandit_scripts/show_bandit.py -a ppo -e noisy_static --show
+The simulation config must be a python file.
+Let's break down the various components of the simulation config.
+
+The first thing you need to do is define a function that is captured by the simulation ingredient.
+The name of the function itself doesn't matter, but we use `config` here for clarity.
+
+``` python
+from simulation import simulation_ingredient  # Import the simulation ingredient
+
+@simulation_ingredient.config  # Tag as the simulation config
+def config():
+    ...
+```
+
+We use the simulation config to construct the simulation `Environment`.
+As such, the config should specify the inputs to the `Environment` class' initialiser.
+
+``` python
+"""
+    distributor (dict[str, Any]): The config for the query distributor.
+    entities (list[dict[str, Any]]): The configs for each group of entities.
+    nepisodes (int): How many episodes to run.
+    ntimesteps (int): How many timesteps to run each episode for.
+"""
+```
+
```

Comment:
ok, so we got distributor and entities... those clearly should be different configs.
How about `ntimesteps ` and  `nepisodes` - shouldn't they go to a separate config called e.g. experiment?

Reply:
I'm glad you posted this actually! I've been thinking about it a bit more, and I think I now disagree with even putting the distributor in its own config. The distributor is a part of the simulation. It's not used elsewhere. Why would we move it to be its own thing in that case? `ntimesteps` and `nepisodes` to me are the same. Those are attributes of the simulation. Do you agree?